lib/core/msdf-sys/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+extern crate wasm_bindgen;
+#[macro_use] extern crate slice_as_array;
+
+use wasm_bindgen::prelude::*;
+use js_sys::*;
+
+const MAX_MSDF_SIZE : usize = 32;
+pub const MSDF_CHANNELS_COUNT : usize = 3;
+const MSDF_DATA_SIZE : usize = MAX_MSDF_SIZE*MAX_MSDF_SIZE*MSDF_CHANNELS_COUNT;
+
+pub struct FontHandle {
+    handle: JsValue
+}
+
+pub struct MSDF {
+    pub width  : usize,
+    pub height : usize,
+    pub data   : [f32;MSDF_DATA_SIZE]
+}
+
+pub struct MSDFParameters {
+    width                         : usize,
+    height                        : usize,
+    edge_coloring_angle_threshold : f64,
+    range                         : f64,
+    scale                         : (f64, f64),
+    translate                     : (f64, f64),
+    edge_threshold                : f64,
+    overlap_support               : bool
+}
+
+#[wasm_bindgen(module = "msdfgen_wasm.js")]
+extern {
+
+fn ccall(
+    name        : &str,
+    return_type : &str,
+    types       : js_sys::Array,
+    values      : js_sys::Array
+) -> JsValue;
+
+fn getValue(address: usize, a_type: &str) -> JsValue;
+
+fn _msdfgen_maxMSDFSize() -> usize;
+
+fn _msdfgen_generateMSDF(
+    width                           : usize,
+    height                          : usize,
+    font_handle                     : JsValue,
+    unicode                         : u32,
+    edge_coloring_angle_threshold   : f64,
+    range                           : f64,
+    scale_x                         : f64,
+    scale_y                         : f64,
+    translate_x                     : f64,
+    translate_y                     : f64,
+    edge_threshold                  : f64,
+    overlap_support                 : bool
+) -> usize;
+
+fn _msdfgen_freeFont(font_handle: JsValue);
+
+}
+
+fn make_js_array_from_data_view(data: &[u8]) -> Uint8Array {
+    unsafe {
+        return Uint8Array::view(data)
+    }
+}
```

Comment:
Wait, this is unsafe function, right? When I was talking about refactoring out such functions, I meant - refactor the smallest portion that is actually safe. Unless I'm missing something, this one is still very unsafe as it may gets invalidated by alloc, right?